File: implementations/java-springboot/harness/test/testdata/aggregate-no-public-setters/good/account/domain/Account.java
package com.example.accountservice.account.domain;

public class Account {
    private String accountId;
    private AccountStatus status;

    private Account() {
    }

    public static Account create(String ownerId) {
        return new Account();
    }

    public void suspend() {
        this.status = AccountStatus.SUSPENDED;
    }

    public String getAccountId() {
        return accountId;
    }
}

File: implementations/kotlin-springboot/harness/test/testdata/aggregate-no-public-setters/bad-public-var/domain/Account.kt
package com.example.accountservice.account.domain

class Account private constructor() {
    var accountId: String = ""
        private set

    // Public setter — allows external code to assign directly, bypassing domain methods, e.g. account.status = "CLOSED"
    var status: String = "ACTIVE"

    companion object {
        fun create(): Account = Account().apply { this.accountId = "acc-1" }
    }

    fun suspend() {
        status = "SUSPENDED"
    }
}
